identity-account-storage/src/storage/stronghold.rs

```diff
@@ -365,6 +407,56 @@ async fn sign_ed25519(vault: &Vault<'_>, payload: Vec<u8>, location: &KeyLocatio
   Ok(Signature::new(signature.into()))
 }
 
+async fn diffie_hellman(
+  vault: &Vault<'_>,
+  private_key: &KeyLocation,
+  public_key: [u8; X25519::PUBLIC_KEY_LENGTH],
+  shared_key: &KeyLocation,
+) -> Result<()> {
+  let diffie_hellman: procedures::X25519DiffieHellman = procedures::X25519DiffieHellman {
+    public_key,
+    private_key: private_key.into(),
+    shared_key: shared_key.into(),
+    hint: default_hint(),
+  };
+  vault.execute(diffie_hellman).await.map_err(Into::into)
+}
+
+async fn aead_encrypt(
+  vault: &Vault<'_>,
+  location: &KeyLocation,
+  plaintext: Vec<u8>,
+  associated_data: Vec<u8>,
+) -> Result<EncryptedData> {
+  let nonce: &[u8] = &Aes256Gcm::random_nonce()?;
+  let aead_encrypt: procedures::AeadEncrypt = procedures::AeadEncrypt {
+    cipher: procedures::AeadCipher::Aes256Gcm,
+    associated_data: associated_data.clone(),
+    plaintext,
+    nonce: nonce.to_vec(),
+    key: location.into(),
+  };
```

Comment:
Note that this uses the 32-byte/256-bit shared secret directly. This is basically the "Direct Key Agreement" mode referenced here: https://datatracker.ietf.org/doc/html/draft-madden-jose-ecdh-1pu-04#section-2

For DIDComm we need the "Key Wrapping" mode with the A256KW algorithm, as that is the only one the specification mentions for `X25519` keys: https://identity.foundation/didcomm-messaging/spec/#key-wrapping-algorithms

We either need to be more flexible and allow passing the "Key Wrapping" algorithm or mode to the encrypt/decrypt functions, or hardcode that ECDH-1PU always uses A256KW (less preferred).

Reply:
This is implemented in the `encrypt_cek` function in `libjose`: https://github.com/iotaledger/identity.rs/blob/6c7d5a6b8d9f507839b9c1e92c3efb2609a72819/libjose/src/jwe/encoder.rs#L355

It uses the `aes-kw` feature of `crypto.rs`.

Reply:
Opened https://github.com/iotaledger/stronghold.rs/issues/338 to request A256KW support in Stronghold.